src/redux/productsApi.ts

```diff
@@ -0,0 +1,25 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { apiUrl } from '../lib/constants';
+import { getProductsList } from '../services/getProductsList';
+import { HYDRATE } from 'next-redux-wrapper';
+
+export const productsApi = createApi({
+  reducerPath: 'productsApi',
+  baseQuery: fetchBaseQuery({
+    baseUrl: apiUrl,
+  }),
+  extractRehydrationInfo(action, { reducerPath }) {
+    if (action.type === HYDRATE) {
+      return action.payload[reducerPath];
+    }
+  },
+  endpoints: (builder) => ({
+    getProductsList: builder.query({
```

Comment:
Why do not you use types for endpoint response and request? There is a console error in your app "Error encountered handling the endpoint getProductsList.
              The object returned by `queryFn` has the unknown property status.
              It needs to return an object with either the shape `{ data: <value> }` or `{ error: <value> }` that may contain an optional `meta` property.     
              Object returned was..."